src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/CollectionDataContract.cs

```diff
@@ -68,14 +68,16 @@ public T Value
 
     [DataContract(Namespace = "http://schemas.microsoft.com/2003/10/Serialization/Arrays")]
 #if USE_REFEMIT
-    public struct KeyValue<K, V>
+    public class KeyValue<K, V>
 #else
-    internal struct KeyValue<K, V>
```

Comment:
Changing KeyValue from a struct to a class will have a significant performance hit which you won't see in a micro benchmark. This could place a lot of stress on the heap and cause long GC pauses some time later. It will could also cause significantly slower memory accesses as an array of structs are next to each other in memory. An array of classes will require an extra dereference which will be slower, and you won't benefit from cache locality and on high core machines under sufficient load to context switch, this could be a significant issue.